utils/python/CIME/case_st_archive.py

```diff
@@ -1,24 +1,413 @@
+        names = filename.split('.')
+        datename = names[-2]
+        datenames.append(datename)
+        logger.debug('cpl dateName: %s ' % datename)
+    return datenames
+
+
+###############################################################################
+def get_ninst_info(case, compclass):
+###############################################################################
+
+    if compclass != 'cpl':
+        ninst = case.get_value('NINST_' + compclass.upper())
+    else:
+        ninst = 1
+    ninst_strings = []
+    if ninst is None:
+        ninst = 1
+        for i in range(ninst):
+            if ninst > 1:
+                ninst_strings.append('_' + '%04d' % i)
+            else:
+                ninst_strings.append('')
+
+    logger.debug("ninst and ninst_strings are: %s and %s for %s" %(ninst,ninst_strings,compclass))
+    return ninst, ninst_strings
+
+           
+###############################################################################
+def archive_rpointer_files(case, archive, archive_entry, archive_restdir, 
+                     datename, datename_is_last):
+###############################################################################
+
+    # archive the rpointer files associated with datename 
+    casename = case.get_value("CASE")
+    compname,compclass = archive.get_entry_info(archive_entry)
+    ninst, ninst_strings = get_ninst_info(case, compclass)
+
+    if (datename_is_last):
+        # Copy of all rpointer files for latest restart date
+        rundir = case.get_value("RUNDIR")
+        rpointers = glob.glob(os.path.join(rundir,'rpointer.*'))
+        for rpointer in rpointers:
+            shutil.copy(rpointer,os.path.join(archive_restdir,os.path.basename(rpointer)))
+    else:
+        # Generate rpointer file(s) for interim restarts for the one datename and each 
+        # possible value of ninst_strings
+        if case.get_value('DOUT_S_SAVE_INTERIM_RESTART_FILES'):
+
+            # parse env_archive.xml to determine the rpointer files
+            # and contents for the given archive_entry tag
+            rpointer_items = archive.get_rpointer_contents(archive_entry)
+
+            # loop through the possible rpointer files and contents
+            for rpointer_file,rpointer_content in rpointer_items:
+
+                # put in a temporary setting for ninst_strings if they are empty
+                # in order to have just one loop over ninst_strings below
+                if rpointer_content is not 'unset':
+                    if not ninst_strings:
+                        ninst_strings = ["empty"]
+                for ninst_string in ninst_strings:
+                    if ninst_string == 'empty':
+                        ninst_string = ""
+                    subs = dict()
+                    subs['$CASE'] = casename
+                    subs['$DATENAME'] = datename
+                    subs['$NINST_STRING'] = ninst_string
+                    for key in subs.keys():
```

Comment:
for key, value in [ ('$CASE', casename), ('$DATENAME', datename), ('$NINST_STRING', ninst_string) ]: